index.bs

```diff
@@ -368,7 +364,27 @@ The <dfn method for="XRSession">end()</dfn> method provides a way to manually sh
 
 </div>
 
-<div class="algorithm" data-algorithm="request-frame-of-reference">
+Each {{XRSession}} has an <dfn>active render state</dfn> which is a new {{XRRenderState}}, a <dfn>pending render state</dfn> which is a new {{XRRenderState}}, and a <dfn>list of pending render state promises</dfn>, which is initially empty.
+
+The <dfn attribute for="XRSession">renderState</dfn> attribute returns the {{XRSession}}'s [=active render state=].
+
+<div class="algorithm" data-algorithm="update-render-state">
+
+When the <dfn method for="XRSession">updateRenderState(|newState|)</dfn> method is invoked, the user agent MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
+
+  1. Let |session| be the target {{XRSession}}.
+  1. Let |pendingState| be |session|'s [=pending render state=].
+  1. If |newState|'s {{XRRenderStateInit/depthNear}} value is set, set |pendingState|'s {{XRRenderState/depthNear}} to |newState|'s {{XRRenderStateInit/depthNear}}.
+  1. If |newState|'s {{XRRenderStateInit/depthFar}} value is set, set |pendingState|'s {{XRRenderState/depthFar}} to |newState|'s {{XRRenderStateInit/depthFar}}.
+  1. If |newState|'s {{XRRenderStateInit/baseLayer}} is set, set |pendingState|'s {{XRRenderState/baseLayer}} to |newState|'s {{XRRenderStateInit/baseLayer}}.
+  1. If |session|'s [=processing frame=] flag is <code>false</code> and |session|'s [=list of animation frame callbacks=] is empty, copy |session|'s [=pending render state=] to |session|'s [=active render state=] and [=/Resolve=] |promise|.
```

Comment:
I am confused by the need for this step.  Does it imply that doing the following:

> session.requestAnimationFrame();
session.updateRenderState(…);
let depthNear = session.renderState.depthNear;

would return a different value for depthNear than: 

> session.updateRenderState(…);
session.requestAnimationFrame();
let depthNear = session.renderState.depthNear;

If so, it seems unexpected to say the least.

To make things more consistent, it seems like we want the requested render state to "lock in" right before Session.RAF callbacks are executed, and therefore always after the current JavaScript execution is finished.